Notification improvements

jojoob opened the pull request:
- Use Messaging API instead of email_to_user() function
- Change notification settings and behaviour.
  - Replace global 'sendmailtoteacher' setting (checkbox) by 'notifycoursebased' (checkbox): default for new instances
  - Replace global 'sendmailtomanager' setting (checkbox) by 'notifyglobal' (multi-select): user based selection (users with 'manageapplications' capability in system context)
  - Replace instance notification setting (checkbox) by user based selection (multi-select): enrolled users with 'manageapplications' capability in course context
  - Upgrade process:
    - if 'sendmailtoteacher' was true, set 'notifycoursebased' true
    - if 'sendmailtomanager' was true, set 'notifyglobal' to "Everyone who can 'Manage apply enrolment'", "Nobody" otherwise
    - if instance based notification setting was set to true, set new user based setting to "Everyone who can 'Manage apply enrolment'", "Nobody" otherwise

jojoob commented:
Mind the version specification in upgrade.php when integrating this and when rising the version number for a new release.

Flottertotte commented:
Hey, looks cool. 
I want to release a new version (Version 3.1-b). What do you mean with: 

> Mind the version specification in upgrade.php when integrating this and when rising the version number for a new release.

Flotter

Flottertotte commented:
Oh I think I know now. Do you mean besides updating $plugin->release
not to forget to also update $plugin->version  
?

Thanks,
Flotter

jojoob commented:
I mean the version specification in [upgrade.php line 70](https://github.com/sudile/moodle-enrol_apply/blob/e156f0ebcedecd53fd4b9beeeecf6f0b8c2ab16d/db/upgrade.php#L70).

If you change $plugin->version in version.php you should also change the value in the upgrade.php and test the upgrade process.

jojoob commented:
I've updated the branch and rebased it to containing only two commits. Since there was no version change at the master branch you can just merge this branch.
